src/app/digital/public/api/impl/DigitalComponent.ts

```diff
@@ -16,4 +18,32 @@ export class DigitalComponentImpl extends ComponentImpl<
     public override get info(): DigitalComponentInfo {
         return new DigitalComponentInfoImpl(this.circuit, this.kind);
     }
+
+    public override firstAvailable(portGroup: Port["group"]): DigitalPort | undefined {
+        if (!this.info.portGroups.includes(portGroup))
+            return undefined; // Invalid port group for the component
+        
+        const ports = this.internal.getPortsForComponent(this.id);
+
+        // Find out if the portGroup is of type input or output
+        const isInputType = this.info.inputPortGroups.includes(portGroup);
+        const isOutputType = this.info.outputPortGroups.includes(portGroup);
+
+        const firstAvailableHelper = (id: string) => {
+            const portObject = this.internal.getPortByID(id)!;
```

Comment:
nit: do the `unwrap` up here, just `const portObject = this.internal.getPortByID(id).unwrap();` (you can remove the exclamation point now)